front/app/api/analyses/useAddAnalysis.ts

```diff
@@ -0,0 +1,30 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { CLErrors } from 'typings';
+import fetcher from 'utils/cl-react-query/fetcher';
+import areaKeys from './keys';
+import { IAnalysis, IAddAnalysis } from './types';
+
+const addAnalysis = async (requestBody: IAddAnalysis) =>
+  fetcher<IAnalysis>({
+    path: '/analyses',
+    action: 'post',
+    body: {
+      analysis: {
+        project_id: requestBody.projectId,
+        phase_id: requestBody.phaseId,
+        custom_field_ids: requestBody.customFieldIds,
+      },
+    },
+  });
+
+const useAddAnalysis = () => {
+  const queryClient = useQueryClient();
+  return useMutation<IAnalysis, CLErrors, IAddAnalysis>({
+    mutationFn: addAnalysis,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: areaKeys.lists() });
```

Comment:
So you copy paste from areas? :)

Reply:
Will fix this later as to not switch branches again. Taking a note!